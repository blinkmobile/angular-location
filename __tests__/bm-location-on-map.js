/* @flow */
'use strict'

// see: https://docs.angularjs.org/guide/unit-testing
// see: http://facebook.github.io/jest/docs/tutorial-jquery.html

const ng = require('angular')
require('angular-mocks')
require('ngmap')

require('../lib/index.js')

beforeEach(ng.mock.module('bmLocation'))

let $compile, $rootScope
beforeEach(ng.mock.inject((_$compile_, _$rootScope_) => {
  $compile = _$compile_
  $rootScope = _$rootScope_
}))

test('no attributes', () => {
  const html = '<bm-location-on-map></bm-location-on-map>'

  const element = $compile(html)($rootScope)
  $rootScope.$digest()
  expect(element.html()).toMatchSnapshot()
})

test('$rootScope.googleMapsApiKey, [coords]', () => {
  $rootScope.googleMapsApiKey = 'secret'
  const coords = JSON.stringify({ latitude: -35, longitude: 150 })

  const html = `
  <bm-location-on-map
    coords='${coords}'
  ></bm-location-on-map>
  `

  const element = $compile(html)($rootScope)
  $rootScope.$digest()
  expect(element.html()).toMatchSnapshot()
})

test('$rootScope.googleMapsApiKey, [coords] [disabled]', () => {
  $rootScope.googleMapsApiKey = 'secret'
  const coords = JSON.stringify({ latitude: -35, longitude: 150 })

  const html = `
  <bm-location-on-map
    coords='${coords}'
    disabled="disabled"
  ></bm-location-on-map>
  `

  const element = $compile(html)($rootScope)
  $rootScope.$digest()
  expect(element.html()).toMatchSnapshot()
})

test('$rootScope.googleMapsApiKey, [coords] [readonly]', () => {
  $rootScope.googleMapsApiKey = 'secret'
  const coords = JSON.stringify({ latitude: -35, longitude: 150 })

  const html = `
  <bm-location-on-map
    coords='${coords}'
    readonly="readonly"
  ></bm-location-on-map>
  `

  const element = $compile(html)($rootScope)
  $rootScope.$digest()
  expect(element.html()).toMatchSnapshot()
})

test('$rootScope.googleMapsApiKey, [coords] [disabled] [readonly]', () => {
  $rootScope.googleMapsApiKey = 'secret'
  const coords = JSON.stringify({ latitude: -35, longitude: 150 })

  const html = `
  <bm-location-on-map
    coords='${coords}'
    disabled="disabled"
    readonly="readonly"
  ></bm-location-on-map>
  `

  const element = $compile(html)($rootScope)
  $rootScope.$digest()
  expect(element.html()).toMatchSnapshot()
})
