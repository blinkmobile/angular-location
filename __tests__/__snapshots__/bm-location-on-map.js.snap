exports[`test $rootScope.googleMapsApiKey, [coords] [disabled] [readonly] 1`] = `
"
  <!-- ngIf: $ctrl.disabled --><bm-static-location-on-map ng-if=\"$ctrl.disabled\" coords=\"$ctrl.coords\" class=\"ng-scope ng-isolate-scope\">
  <figure>
    <img ng-src=\"https://maps.googleapis.com/maps/api/staticmap?key=secret&amp;center=-35%2C150&amp;markers=color%3Ared%7C-35%2C150&amp;scale=2&amp;size=300x300&amp;zoom=5\" title=\"map with center at -35 latitude, 150 longitude\" height=\"300\" width=\"300\" src=\"https://maps.googleapis.com/maps/api/staticmap?key=secret&amp;center=-35%2C150&amp;markers=color%3Ared%7C-35%2C150&amp;scale=2&amp;size=300x300&amp;zoom=5\">
  </figure>
</bm-static-location-on-map><!-- end ngIf: $ctrl.disabled -->
  <!-- ngIf: !$ctrl.disabled -->
"
`;

exports[`test $rootScope.googleMapsApiKey, [coords] [disabled] 1`] = `
"
  <!-- ngIf: $ctrl.disabled --><bm-static-location-on-map ng-if=\"$ctrl.disabled\" coords=\"$ctrl.coords\" class=\"ng-scope ng-isolate-scope\">
  <figure>
    <img ng-src=\"https://maps.googleapis.com/maps/api/staticmap?key=secret&amp;center=-35%2C150&amp;markers=color%3Ared%7C-35%2C150&amp;scale=2&amp;size=300x300&amp;zoom=5\" title=\"map with center at -35 latitude, 150 longitude\" height=\"300\" width=\"300\" src=\"https://maps.googleapis.com/maps/api/staticmap?key=secret&amp;center=-35%2C150&amp;markers=color%3Ared%7C-35%2C150&amp;scale=2&amp;size=300x300&amp;zoom=5\">
  </figure>
</bm-static-location-on-map><!-- end ngIf: $ctrl.disabled -->
  <!-- ngIf: !$ctrl.disabled -->
"
`;

exports[`test $rootScope.googleMapsApiKey, [coords] [readonly] 1`] = `
"
  <!-- ngIf: $ctrl.disabled -->
  <!-- ngIf: !$ctrl.disabled --><figure ng-if=\"!$ctrl.disabled\" map-lazy-load=\"https://maps.google.com/maps/api/js\" map-lazy-load-params=\"https://maps.google.com/maps/api/js?key=secret\" class=\"ng-scope\"></figure><!-- end ngIf: !$ctrl.disabled -->
"
`;

exports[`test $rootScope.googleMapsApiKey, [coords] 1`] = `
"
  <!-- ngIf: $ctrl.disabled -->
  <!-- ngIf: !$ctrl.disabled --><figure ng-if=\"!$ctrl.disabled\" map-lazy-load=\"https://maps.google.com/maps/api/js\" map-lazy-load-params=\"https://maps.google.com/maps/api/js?key=secret\" class=\"ng-scope\"></figure><!-- end ngIf: !$ctrl.disabled -->
"
`;

exports[`test no attributes 1`] = `
"
  <!-- ngIf: $ctrl.disabled -->
  <!-- ngIf: !$ctrl.disabled --><figure ng-if=\"!$ctrl.disabled\" map-lazy-load=\"https://maps.google.com/maps/api/js\" map-lazy-load-params=\"https://maps.google.com/maps/api/js?key=undefined\" class=\"ng-scope\"></figure><!-- end ngIf: !$ctrl.disabled -->
"
`;
